install.sh

```diff
@@ -615,5 +685,76 @@ fi
 
 }
 
+function register_sandcats_name() {
+  # We allow environment variables to override some details of the
+  # Sandcats service, so that during development, we can test against
+  # a non-production Sandcats service.
+  SANDCATS_BASE_DOMAIN="${OVERRIDE_SANDCATS_BASE_DOMAIN:-sandcats.io}"
+  SANDCATS_API_BASE="${OVERRIDE_SANDCATS_API_BASE:-https://sandcats.io}"
+  SANDCATS_CURL_PARAMS="${OVERRIDE_SANDCATS_CURL_PARAMS:-}"
+
+  echo "Press enter to accept a random name, or choose your own (alphanumeric, max len 20)."
+  echo "Type the word none to skip this step."
+  DESIRED_SANDCATS_NAME=$(prompt "What *.${SANDCATS_BASE_DOMAIN} subdomain would you like?" '')
+  if [ -z "$DESIRED_SANDCATS_NAME" ] ; then
+    DESIRED_SANDCATS_NAME=$(random_sandcats_name)
+  fi
+
+  # If the user really wants none of our sandcats help, then bail out.
+  if [ "none" = "$DESIRED_SANDCATS_NAME" ] ; then
+    return
+  fi
+
+  # Validate the client-side, to avoid problems, against the same
+  # regex that the server is using.
+  [[ $DESIRED_SANDCATS_NAME =~ ^[0-9a-zA-Z]{1,20}$ ]] || register_sandcats_name
```

Comment:
re: capital letters: I downcase them on the server-side, and I figure some people are going to type capital letters, so I figure it's OK to let them type it here, and downcase it later in the process.

re: internal hyphens: I'll deal with hyphens in a separate pull request, I think.

re: minimum length: whatever, land grabs are fun!

re: recursion: nope, it doesn't work; fixed in a follow-up commit.